### PR TITLE
added unicode escaping

### PR DESCRIPTION
**Pull request by sadan4**
Unicode escaping now works in config theme files!! 

use `\uxxxx` to use a Unicode value

### CODE REVIEW

**Comment by keli5** [comment]
Doesn't work with extended Unicode e.g. \u1f386 as a prompt character makes the character come out "I6"

**Review by keli5 — CHANGES_REQUESTED** [review]
see comment

**Review by sadan4 — COMMENTED** [review]
fixed

File: shellutil/settingsutil.hpp [hunk 1]
```diff
@@ -98,6 +98,17 @@ inline std::string readableToEscape(const std::string &input) {
             i += 6;
             int codepoint = std::stoi(chars, nullptr, 16);
             result << getUnicodeChar(codepoint);
+        } else if (input[i] == '\\' && input[i + 1] == 'U') {
```
Comment (keli5) [inline shellutil/settingsutil.hpp:101]: Could this if statement be merged with the one above? Something like `input[i] == '\\' && (input[i + 1] == 'U' || input[i + 1] == 'u')`

**Review by keli5 — CHANGES_REQUESTED** [review]

File: shellutil/settingsutil.hpp (hunk 1 above)
Comment (sadan4) [reply shellutil/settingsutil.hpp:101]: each uses a different number of characters, they could but then you would have to do something like `\u001234` for a four digit unicode character

your choice

Comment (keli5) [reply shellutil/settingsutil.hpp:101]: I figured that's what you were supposed to do anyways based on the last commit message, think it would make more sense and be simpler code to do it that way

Comment (sadan4) [reply shellutil/settingsutil.hpp]: done

**Comment by keli5** [comment]
probably looks good
